app/(dashboard)/item/[id]/page.tsx

```diff
@@ -25,7 +19,8 @@ const DisplayItemDetails = async ({ params }: { params: { id: string } }) => {
       throw new Error('Error fetching data');
     } else {
       const donorEmail = item.profiles.email;
-      const donerId: string = item.profiles?.id;
+      const donorId: string = item.profiles?.id;
+      const donorName = item.profiles.username;
       const title = item.item_name;
```

Comment:
Do we need these? From my read it looks like we're just using them as props and we already pass the entire `item` object so why don't we just attribute all these values inside the component.

The new component could probably have a much slimmer declaration:
```js
       <DisplayItemDetailsClient
          item={item}
          user={user!}
          canMessage={canMessage}
        />
```

Reply:
I ran into a problem after removing the `donorId`, `donorEmail`, and `donorName` . I'm getting a TypeScript error saying that profiles does not exist on the item object.

Without these props, TypeScript can’t confirm that profiles will be available.

I can remove only 'title' from props without facing type error.

Should I leave  `donorId`, `donorEmail`, and `donorName` and remove 'title'?

Reply:
you're right - `item.profiles` doesn't exist. We can replace `donorName` and `donorId` by `item.username` and `item.donated_by` though. That leaves just `donorEmail` which is still much leaner so let's try that :)

Reply:
I removed `donorId` and `title`